src/components/Articles.jsx

```diff
@@ -1,20 +1,30 @@
 import { API } from "../api";
 
 import { useState, useEffect } from "react";
+import { useSearchParams } from "react-router-dom";
 
 import ArticleCard from "./ArticleCard";
 import Loading from "./Loading";
 
 export default function Articles()
 {
+    let [searchParams, setSearchParams] = useSearchParams();
+    const topic = searchParams.get("topic");
+
     const [articles, setArticles] = useState([]);
     const hasLoaded = !!articles.length;
 
     useEffect(() => {
         const fetchArticles = async () => {
             try
             {
-                const response = await API.get("/articles");
+                let URL = "/articles?";
+                if (topic)
+                {
+                    URL += `topic=${topic}`;
+                }
+
+                const response = await API.get(URL);
```

Comment:
This approach works to achieve doing this API call with or without the extra queries. Nice use of searchParams to retrieve the topic.
This would be improved by separating this API call into a function in your api file, which would separate our concerns a bit more and keep this logic out of the component.